collect_app/src/androidTest/java/org/odk/collect/android/feature/external/InstanceUploadActionTest.kt

```diff
@@ -1,40 +1,68 @@
 package org.odk.collect.android.feature.external
 
+import android.content.Context
 import android.content.Intent
+import android.provider.BaseColumns._ID
+import androidx.test.core.app.ApplicationProvider
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
 import org.junit.Rule
 import org.junit.Test
 import org.junit.rules.RuleChain
 import org.junit.runner.RunWith
 import org.odk.collect.android.external.InstancesContract
-import org.odk.collect.android.instancemanagement.send.InstanceUploaderActivity
+import org.odk.collect.android.support.TestDependencies
+import org.odk.collect.android.support.pages.FormEntryPage
 import org.odk.collect.android.support.pages.OkDialog
 import org.odk.collect.android.support.rules.CollectTestRule
 import org.odk.collect.android.support.rules.TestRuleChain
+import org.odk.collect.android.utilities.ApplicationConstants
 
 @RunWith(AndroidJUnit4::class)
 class InstanceUploadActionTest {
 
-    val collectTestRule = CollectTestRule()
+    private val rule = CollectTestRule(useDemoProject = false)
+    private val context = ApplicationProvider.getApplicationContext<Context>()
+    private val testDependencies = TestDependencies()
 
     @get:Rule
-    val rule: RuleChain = TestRuleChain.chain()
-        .around(collectTestRule)
+    val chain: RuleChain = TestRuleChain.chain(testDependencies)
+        .around(rule)
 
     @Test
-    fun whenInstanceDoesNotExist_showsError() {
-        val instanceIds = longArrayOf(11)
-        instanceUploadAction(instanceIds)
+    fun canUploadInstanceToDifferentServer() {
```

Comment:
This is not the same as sending to a different server based on submission uri from the instances db right? You can trigger uploading with a custom URL? The title might be misleading, it would be good to either add a comment or improve it to distinguish those two cases.